spec/unit/resource/chocolatey_installer_spec.rb

```diff
@@ -0,0 +1,109 @@
+#
+# Copyright:: Copyright (c) Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "spec_helper"
+
+begin
+  require "chef-powershell"
+rescue LoadError
+end
+
+describe Chef::Resource::ChocolateyInstaller do
+  include RecipeDSLHelper
+
+  let(:resource) { Chef::Resource::ChocolateyInstaller.new("fakey_fakerton") }
+  let(:config) do
+    <<-CONFIG
+      <?xml version="1.0" encoding="utf-8"?>
+      <chocolatey xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+        <config>
+          <add key="containsLegacyPackageInstalls" value="true" description="Install has packages installed prior to 0.9.9 series." />
+        </config>
+        <sources>
+          <source id="chocolatey" value="https://chocolatey.org/api/v2/" disabled="false" bypassProxy="false" selfService="false" adminOnly="false" priority="0" />
+        </sources>
+        <features>
+          <feature name="checksumFiles" enabled="true" setExplicitly="false" description="Checksum files when pulled in from internet (based on package)." />
+        </features>
+        <apiKeys />
+      </chocolatey>
+    CONFIG
+  end
+
+  # we save off the ENV and set ALLUSERSPROFILE so these specs will work on *nix and non-C drive Windows installs
+  before(:each) do
+    @original_env = ENV.to_hash
+    ENV["ALLUSERSPROFILE"] = "C:\\ProgramData"
+  end
+
+  after(:each) do
+    ENV.clear
+    ENV.update(@original_env)
+  end
+
+  describe "basic chocolatey settings" do
+    it "has a resource name of :chocolatey_installer" do
+      expect(resource.resource_name).to eql(:chocolatey_installer)
+    end
+
+    it "sets the default action as :install" do
+      expect(resource.action).to eql([:install])
+    end
+
+    it "supports :install and :uninstall actions" do
+      expect { resource.action :install }.not_to raise_error
+      expect { resource.action :uninstall }.not_to raise_error
+    end
+  end
+
+  describe "installing chocolatey" do
+    it "can install Chocolatey with parameters" do
+      expect { resource.action :install }.not_to raise_error
+    end
+
+    it "returns false if a chocolatey install cannot be found" do
```

Comment:
The `it` here is not the same as the `it` above.  This tests `fetch_choco_installer` specifically, not the resource.

Reply:
corrected